### ISSUE TITLE
13.2 gives mysterious error on code that linted fine with 13.1

### ISSUE THREAD

**Issue by rbirkgit** [angular-eslint/angular-eslint#984]
**Description and reproduction of the issue**

The following code linted fine with 13.1 but with 13.2 I now get a cryptic error:

`error  Parsing error: Parser Error: Unexpected token '{' at column 3 in [consumed resources are at {{ ${percent} | formatPercentLocale }}] `

```
const percent = 0.123;
@Component({ template: `<p>consumed resources are at {{ ${percent} | formatPercentLocale }}</p>` })
class TestPipeComponent {}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/template-parser`  | `13.2.0` |
| `ESLint`                           | `8.13.0` |
| `node`                             | `16.14.2` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

Angular CLI: 13.3.2
Node: 16.14.2
Package Manager: yarn 3.1.1
OS: win32 x64

Angular: 13.3.2
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1303.2
@angular-devkit/build-angular   13.3.2
@angular-devkit/core            13.3.2
@angular-devkit/schematics      13.3.2
@schematics/angular             13.3.2
ng-packagr                      13.3.1
rxjs                            7.5.5
typescript                      4.6.3
```

- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
The reason this is happening in version `13.2` of angular-eslint is that we now respect parse errors from the `@angular/compiler` and surface them to you.

Your template contents, when viewed statically in isolation, cannot be parsed by the `@angular/compiler`. The only way your template can be evaluated is if it is executed, which does not happen as part of the linting lifecycle (but does when your app compiles with ngc).

![image](https://user-images.githubusercontent.com/900523/163360519-dfc7fee1-a00c-4354-bbb1-0a89cc3feaff.png)

This is a tricky one, because we definitely do want to surface parse errors to users, so reverting is not what we want to do.

I think we will simply have to add a parserOption to allow users to suppress parse errors if they want to do things like you have here which would required runtime evaluation of the template.

Please note, however, that there may be cases where the produced AST does not have enough data in it (because of the parse errors) in order for lint rules to run completely accurately, so bear that in mind when choosing to suppress parse errors (I can't comment on the exact nature of fault tolerance in the AST building within the angular/compiler it's just a general disclaimer).

**Comment by JamesHenry** [comment]
I decided to class this as an unintentional breaking changes, so I have created https://github.com/angular-eslint/angular-eslint/pull/987 which will revert the default behaviour for v13, but allow folks to opt in to respecting parse options, with the intention that this default behaviour will switch in v14 as an intentional breaking change

**Comment by JamesHenry** [comment]
Addressed by #987 will be released as a patch shortly

**Comment by rbirkgit** [comment]
Thanks. I was hoping one could suppress this on case by case in the code, like you do with rules. But that may not work in this case. 

We will plan to rewrite our tests with v14.
